### ISSUE TITLE
Mark this library as not maintained and redirect to the new project

### ISSUE THREAD

**Issue by Spomky** [Spomky-Labs/jose#297]
This library and the associated Symfony bundle will not be maintained anymore by the end of the year 2017.
**Security issues will still be managed until the end of 2019**.

The reason is that the next major version is a complete rewrite handled by [a new organization](https://github.com/web-token).

That new version is still in development but major features are already implemented.

Waiting for 

* [ ] #307 
* [ ] wed-token project to be stable

**Comment by skyzyx** [comment]
Um, is there going to be a suggested replacement package documented in composer.json?

**Comment by Spomky** [comment]
Hi @skyzyx,

Yes, the new project has its own organization. Please refer to https://github.com/web-token
The https://github.com/web-token/jwt-framework is a monolithic repository with all components, but it is preferable to use only the components you need (see all other web-token/jwt-* repositories).

That new project is quite stable, but I need to fix some issues before its first release.
ETA: end of 2017.

The documentation will be available on https://web-token.spomky-labs.com/

At the moment this package is still maintained. I will deprecate it only when the new one will be ready.

**Comment by schodemeiss** [comment]
Does this 7.0 version work with PHP 7.2 or is the newer "web-token" package the one to run with for PHP 7.2 support?

I assume there isn't going to be a smooth migration path between the two projects?

**Comment by Spomky** [comment]
The [last tests show no error](https://travis-ci.org/Spomky-Labs/jose/builds/286689939) on PHP 7.2 (and 7.3-dev).
There are some similarities between the two projects and the migration from one project to another is not so difficult from my POV.
I updated [another project that uses the symfony bundle](https://github.com/Spomky-Labs/lexik-jose-bridge/commit/b8c37264bd5869b9ae53ecacb620d3ca3f6a9c75).

The big changes come from the way you forge tokens: Computation and Serialization steps are now decoupled.

Before:
```php
$jwt = $jwt_creator->sign($payload, $protectedHeader, $signatureKey);
```

After

```php
$jwt = $jwsBuilder
    ->create()
    ->withPayload($payload)
    ->addSignature($signatureKey, $protectedHeader)
    ->build();
$token = $compactSerializer->serialize($jwt, 0);
```

I have not finished the documentation, but [major feature pages are already written](https://web-token.spomky-labs.com/).
Feedbacks are welcome (you can add a comment directly on the documentation).

**Comment by schodemeiss** [comment]
So for my PHP 7.2 support, should I continue to use this library for now until the newer lib has a stable release?

I recall there were some issues with some packages upstream not support 7.2?

**Comment by Spomky** [comment]
You are right, there were some issues with class names in conflict with PHP reserved names.
Everithing is fixed now and you should not encounter any trouble on PHP 7.2+.

At the moment please continue to use this library. The new project is almost ready, but I still have some modifications to perform.

**Comment by Spomky** [comment]
Hi @skyzyx, @schodemeiss 

The new project is now stable enough. I will write an "how-to migrate" to allow you to easily update your applications.
This documentation will be available on the new project. I will add a link here when done.
See also #307.

**Comment by Spomky** [comment]
Dear all,

I have just release the first stable release of the [new JWT framework](https://github.com/web-token/jwt-framework).
I realized that, even if the documentation is not clear, this library is commonly used in projects that need JWE support.
That is why I decided to postpone its end of life. I will deprecate it by the end of 2018 with a support for security issues until the end of 2020.
I hope this will help you to migrate to that new framework a smooth way.
